niceml/filechecksumprocessors/zippedcsvtoparqprocessor.py

```diff
@@ -0,0 +1,184 @@
+            force: bool: Force the generation of batches even if no files have changed
+
+        Returns:
+            A list of batches, each batch is a dictionary with one key `inputs`
+            and the value is a list of file paths
+        """
+        if not force:
+            input_file_list = [
+                file_name
+                for file_name, changed in changed_files_dict["inputs"].items()
+                if changed
+            ]
+        batches = []
+        for batch_pos in range(0, len(input_file_list), self.batch_size):
+            batches.append(
+                {"inputs": input_file_list[batch_pos : batch_pos + self.batch_size]}
+            )
+        return batches
+
+    def process(self, batch: Dict[str, List[str]]) -> Dict[str, Dict[str, str]]:
+        """
+        The process function takes a batch of files and converts them from CSV to Parquet.
+
+        Args:
+            self: Access the class attributes
+            batch: Dict[str: Pass in the batch of files to be processed
+            List[str]]: Pass in the list of files that are to be processed
+
+        Returns:
+            A dictionary of checksums for each file in `self.output_location`
+            (key = `outputs`) and `self.input_location` (key = `inputs`)
+        """
+        checksums = defaultdict(dict)
+
+        with open_location(self.input_location) as (input_file_system, input_root):
+            for zip_file in tqdm(
+                batch["inputs"], desc="Extract zip files of current batch"
+            ):
+                with input_file_system.open(zip_file) as opened_zip_file:
+                    checksums["inputs"][zip_file] = input_file_system.checksum(zip_file)
+
+                    zf = zipfile.ZipFile(opened_zip_file)
+                    csv_files = zf.namelist()
+                    for csv_file in csv_files:
+                        with zf.open(csv_file, mode="r") as opened_csv_file:
+                            df = pd.read_csv(
+                                opened_csv_file,
+                                sep=self.csv_seperator,
+                                low_memory=False,
+                            )
+                        parq_name = basename(splitext(csv_file)[0]) + ".parq"
+                        output_df_location = join_location_w_path(
+                            self.output_location, parq_name
+                        )
+                        with open_location(output_df_location) as (
+                            output_file_system,
+                            output_root,
+                        ):
+                            write_parquet(
+                                df, output_root, file_system=output_file_system
+                            )
+                            checksums["outputs"][
+                                output_root
+                            ] = output_file_system.checksum(output_root)
```

Comment:
Change `output_file_system.checksum` into `md5_from_file`

Reply:
Done in [cc88d45](https://github.com/codecentric-oss/niceml/pull/44/commits/cc88d45d9db160ca1d28c704ec762f3c6c8024fc)